include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -998,16 +1233,17 @@ class operational_domain_impl
  * @return The operational domain of the layout.
  */
 template <typename Lyt, typename TT>
-operational_domain operational_domain_grid_search(const Lyt& lyt, const std::vector<TT>& spec,
-                                                  const operational_domain_params& params = {},
-                                                  operational_domain_stats*        stats  = nullptr)
+auto operational_domain_grid_search(const Lyt& lyt, const std::vector<TT>& spec,
```

Comment:
Since these are API functions, I would refrain from using `auto` as the return type here to provide more information to the users. You may use `operational_domain<parameter_point, operational_status>`.

Reply:
Same for the other operational domain functions

Reply:
I agree! ``auto`` makes it harder to read. Thank you.